api/albums-router.js

```diff
@@ -18,70 +21,72 @@ router.use((req, res, next) => {
   }
 });
 
-router.post('/memberships', async(req, res) => {
+router.post("/memberships", async (req, res) => {
   const { shareToken } = req.body;
   const userToken = req.user.token;
   const userId = req.user.profile.id;
 
-  console.log('who is user?', req.user);
+  console.log("who is user?", req.user);
 
   try {
     const album = await joinAlbum(userToken, userId, shareToken);
     res.json({ album });
   } catch (err) {
-    console.log('ERRRRRRR', err);
+    console.log("ERRRRRRR", err);
     res.status(500).json(err);
   }
 });
 
-router.get('/:albumId/share-token', async(req, res) => {
+router.post("/:albumId/share-token", async (req, res) => {
   const { albumId } = req.params;
   const userToken = req.user.token;
   const userId = req.user.profile.id;
+  const shareUser = req.body.shareUser;
   try {
-    const album = await shareAlbum(userToken, userId, albumId);
+    const album = await shareAlbum(userToken, userId, albumId, shareUser);
```

Comment:
`shareUser` should be an email address but we can discuss other identifier options